pkg/virtualization/core/storage/service/ImageManagementService.go

```diff
@@ -173,3 +173,57 @@ func (ims *ImageManagementService) CreateDisk(settings *disk.VirtualHardDiskSett
 	defer job.Close()
 	return job.WaitForJobCompletion(result.ReturnValue, -1)
 }
+
+func (ims *ImageManagementService) GetVirtualHardDiskConfig(path string) (size uint64,
+	blockSize uint32,
+	lSectorSize uint32,
+	pSectorSize uint32,
+	format uint16,
+	err error) {
+	method, err := ims.GetWmiMethod("GetVirtualHardDiskSettingData")
+	if err != nil {
+		return
+	}
+	defer method.Close()
+
+	inparams := wmi.WmiMethodParamCollection{}
+	inparams = append(inparams, wmi.NewWmiMethodParam("Path", path))
+
+	outparams := wmi.WmiMethodParamCollection{}
+	outparams = append(outparams, wmi.NewWmiMethodParam("SettingData", nil))
+	outparams = append(outparams, wmi.NewWmiMethodParam("Job", nil))
+
+	result, err := method.Execute(inparams, outparams)
+	if err != nil {
+		return
+	}
+
+	if !(result.ReturnValue == 4096 || result.ReturnValue == 0) {
+		err = errors.Wrapf(errors.Failed, "GetVirtualHardDiskSettingData method failed with [%d]", result.ReturnValue)
```

Comment:
And this failed return numeric value should map with human-readable format. I believe that would be great benefit for long term maintenance perspective. 

For example,

Failed (32768)

Access Denied (32769)

Not Supported (32770)

Status is unknown (32771)

Timeout (32772)

Invalid parameter (32773)

System is in used (32774)

Invalid state for this operation (32775)

Incorrect data type (32776)

System is not available (32777)

Out of memory (32778)

File not found (32779)

Reply:
Same for this comment as well. Let me take this in a separate PR.